qemu/tests/block_hotplug.py

```diff
@@ -121,42 +123,47 @@ def get_new_disk(disk1, disk2):
 
             device_list.append(device)
             if not new_disks:
-                raise exceptions.TestFail("Cannot find new disk after hotplug.")
+                test.fail("Cannot find new disk after hotplug.")
 
             if params.get("need_plug") == "yes":
                 disk = new_disks[0]
             else:
                 disk = new_disks[num]
 
+            session = vm.wait_for_login(timeout=timeout)
+            if params.get("format_disk", "no") == "yes":
+                if repeat_time == 0:
+                    error_context.context("Format disk", logging.info)
+                    utils_misc.format_windows_disk(session,
+                                                   disk_index[num],
+                                                   mountpoint=disk_letter[num])
             error_context.context("Check block device after hotplug.",
                                   logging.info)
-            if params.get("pci_test_cmd"):
+            if params.get("disk_op_cmd"):
                 if params.get("os_type") == "linux":
-                    test_cmd = params.get("pci_test_cmd") % (disk, disk)
+                    test_cmd = params.get("disk_op_cmd") % (disk, disk)
+                elif params.get("os_type") == "windows":
+                    test_cmd = params.get("iozone_cmd") % (disk_letter[num],
+                                                           disk_letter[num])
                 else:
-                    test_cmd = re.sub("PCI_NUM", "%s" % (num + 1),
-                                      params.get("pci_test_cmd"))
-                session = vm.wait_for_login(timeout=timeout)
-                s, o = session.cmd_status_output(test_cmd, timeout=disk_op_timeout)
-                session.close()
-                if s:
-                    raise exceptions.TestFail("Check for block device failed "
-                                              "after hotplug, Output: %r" % o)
+                    error_context.context("Has not implemented.", logging.info)
+                    raise NotImplementedError
```

Comment:
could you please also help update find_disk(), which only judge linux and windows ?